### DISCUSSION TITLE
Directory navigation/picker box [Very new user]

### DISCUSSION THREAD

**Discussion by a-jp** [zauberzeug/nicegui#2503] [Q&A]
### Question

Hi,

Considering this as a way to provide a gui for an existing computational tool.  It's run locally, and saves a lot of data.  Multiple runs means users need to save variations of runs in different directories.  Therefore, my first job is to allow the user an easy way to pick the base directory where data (including new sub-directories) are to be written.  How can I do this simply and natively with NiceGUI?  

[Been using NiceGUI for three hours, I have no other third party tools installed, using conda, on WLS2].

Cheers,

**Comment by rodja** [comment]
You may be interested in this discussion about a local file/directory picker: https://github.com/zauberzeug/nicegui/discussions/283

**Reply by a-jp** [reply]
Hi, thanks for the link. I really don't understand that. I'm looking for a class method to just navigate, open and pick a directory, not files. Is this something that can be done easily? Sorry, new user!

**Reply by rodja** [reply]
Yes. You should be able to copy the code from https://github.com/zauberzeug/nicegui/blob/main/examples/local_file_picker/local_file_picker.py and modify it to not pick files but just the right directory.

**Reply by a-jp** [reply]
Thank you for your reply. I appreciate you're trying to help me. For new users, like me, it's odd that opening a file appears as an example rather than core api. And as a user, a new user, not a developer of this framework, it's odd that opening a directory does not appear as core api also, and rather requires detailed knowledge of the api to hack existing code. 

If anyone has hacked the example to provide directory opening, are they able to share that here?

**Reply by a-jp** [reply]
Just wondered if anyone could provide further help on this directory picker question? I'm keen to also learn why this and a file picker are not core api at the moment? That is, is that by design, or just haven't been incorporated as yet into the api?  Thanks

**Reply by falkoschindler** [reply]
I asked ChatGPT to summarize #283:

> The discussion on GitHub regarding the lack of a built-in file picker for directories in NiceGUI is centered around the inherent limitations of browser security. Browsers are designed to not expose the full file system path to web applications for security reasons. This restriction prevents web applications, including those built with NiceGUI, from implementing a directory picker that can access and display the full path of directories on a client's machine. The conversation explores potential workarounds, such as developing a custom file browser or integrating other technologies, but these solutions are complicated and may not be reliable. For a comprehensive understanding, you can refer to the full discussion on [GitHub](https://github.com/zauberzeug/nicegui/discussions/283).

In short: There's an inherent limitation for picking files on the client machine. Therefore we haven't found a universal solution to this problem.

**Reply by a-jp** [reply]
Thanks
